### PR TITLE
Introducing Security Center .Net SDK

### PR DESCRIPTION
**Pull request by chlahav**
## Description
Introducing Security Center .Net SDK from [a new swagger spec](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/security/resource-manager)

[First swagger PR](https://github.com/Azure/azure-rest-api-specs/pull/3144)

[Spec PR](https://github.com/Azure/azure-rest-api-specs/pull/3188)
[Spec PR](https://github.com/Azure/azure-rest-api-specs/pull/3287)
[Spec PR](https://github.com/Azure/azure-rest-api-specs/pull/3372)
[Spec PR](https://github.com/Azure/azure-rest-api-specs/pull/3384)
[Spec PR](https://github.com/Azure/azure-rest-api-specs/pull/3407)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
@chlahav Please update the PR with latest upstream changes

**Comment by chlahav** [comment]
@dsgouda PR was updated with the latest changes

File: src/SDKs/SecurityCenter/Management.SecurityCenter/Microsoft.Azure.Management.SecurityCenter.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <!-- Please do not move/edit code below this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.SecurityCenter</PackageId>
+    <Description>Provides developers with libraries for the updated Azure Security Center platform under Azure Resource manager to view and manage security posture in and outside Azure.</Description>
+    <Version>0.1.0</Version>
```
Comment (dsgouda) [inline src/SDKs/SecurityCenter/Management.SecurityCenter/Microsoft.Azure.Management.SecurityCenter.csproj]: Looks like the SDK is still in preview version. The Nuget package must also be in preview. Preview versioning must start from
`0.9.0-preview`

File: src/SDKs/SecurityCenter/Management.SecurityCenter/generate.ps1 [hunk 2]
```diff
@@ -0,0 +1 @@
+Start-AutoRestCodeGeneration -ResourceProvider "security/resource-manager" -AutoRestVersion "latest"
```
Comment (dsgouda) [inline src/SDKs/SecurityCenter/Management.SecurityCenter/generate.ps1]: Please confirm that the Sdk was generated using this script.
There must also be a SdkInfo_*.cs file and a .txt log file generated, please commit these files

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great for the most part. Please address the comments posted.

File: src/SDKs/SecurityCenter/Management.SecurityCenter/Microsoft.Azure.Management.SecurityCenter.csproj (hunk 1 above)
Comment (chlahav) [reply src/SDKs/SecurityCenter/Management.SecurityCenter/Microsoft.Azure.Management.SecurityCenter.csproj]: changed to `0.9.0-preview`

File: src/SDKs/SecurityCenter/Management.SecurityCenter/generate.ps1 (hunk 2 above)
Comment (chlahav) [reply src/SDKs/SecurityCenter/Management.SecurityCenter/generate.ps1]: the generated files were generated using this script.
I also added the SdkInfo file and the text file

File: src/SDKs/SecurityCenter/Management.SecurityCenter/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Azure Security Center Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Security Center Resources.")]
+
+[assembly: AssemblyVersion("0.1.0.0")]
+[assembly: AssemblyFileVersion("0.1.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/SecurityCenter/Management.SecurityCenter/Properties/AssemblyInfo.cs]: @chlahav update assemblyversion/fileversion to 0.9.0.0

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/SecurityCenter/Management.SecurityCenter/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (chlahav) [reply src/SDKs/SecurityCenter/Management.SecurityCenter/Properties/AssemblyInfo.cs]: @shahabhijeet done

File: src/SDKs/SecurityCenter/Management.SecurityCenter/Microsoft.Azure.Management.SecurityCenter.csproj [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.SecurityCenter</PackageId>
+    <Description>Provides developers with libraries for the updated Azure Security Center platform under Azure Resource manager to view and manage security posture in and outside Azure.</Description>
+    <Version>0.9.0-preview</Version>
+    <AssemblyName>Microsoft.Azure.Management.SecurityCenter</AssemblyName>
+    <PackageTags>management;security center;security;</PackageTags>
+    <PackageReleaseNotes>
+      <![CDATA[
+        This is a public release of the Azure Security Center SDK.
```
Comment (dsgouda) [inline src/SDKs/SecurityCenter/Management.SecurityCenter/Microsoft.Azure.Management.SecurityCenter.csproj]: public preview release

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good, just address the one comment and update the branch and we should be good

**Review by dsgouda — APPROVED** [review]
LGTM subject to CIs passing

**Comment by chlahav** [comment]
@dsgouda All the CI tests passed, if there are no additional comments, can you please merge the CR?
thanks!

**Comment by dsgouda** [comment]
@chlahav Please update the branch with upstream psSdkJson6

**Comment by chlahav** [comment]
@dsgouda branch was updated and all the tests passed

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**